PiwikTracker/Dispatcher.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+internal protocol Dispatcher {
+    associatedtype Dispatchable
+    
+    var userAgent: String? { get set }
+    
+    func send(event: Dispatchable, success: ()->(), failure: (_ shouldContinue: Bool)->());
```

Comment:
In practice, do you think `Dispatchable` will be an `Event` or something else?

I think this this protocol to be really useful `Dispatchable` needs to be constraint by something (not sure what it is). If we can not find what this something is maybe just put `Event` there and refactor later.